code/modules/surgery/blood_filter.dm

```diff
@@ -67,6 +67,7 @@
 		for(var/datum/reagent/chem as anything in target.reagents.reagent_list)
 			if(!length(bloodfilter.whitelist) || (chem.type in bloodfilter.whitelist))
 				target.reagents.remove_reagent(chem.type, min(chem.volume * 0.22, 10))
+	target.adjustToxLoss(-5, forced = TRUE)
```

Comment:
This isn't the right way to do it. Reducing damage done fully depends on the reagent being removed & how much of it is removed.

Some reagents cause burn damage, others cause oxy loose, others toxin damage, so depending on the reagent removed we should repair a specific type of damage accordingly & not just toxin damage in general.

And even then purging toxins doesn't repair the damage it has already done. For e.g. consuming sulfuric acid burns your stomach & destroys your blood so removing it doesn't automatically repair your stomach & purge toxins from your blood & other burned internals.

Overall this isn't realistic & there isn't any downside to providing free repair (unlike meds which comes with their own side effects so we have some nuance & balance there).

If you would have created a new type of machine like a dialysis machine which requires intresting interactions to achieve this effect then this would be more interesting but this isn't it. 👎 from me

Reply:
Isn't the blood filter, literally a portable dialysis machine?

Reply:
I think the biggest downside of this is the fact it requires a second person. For most other treatments, you can just grab a pill, bottle or patch and treat yourself if you're able to. However this literally requires another person, since you can't do it to yourself.

Reply:
> Isn't the blood filter, literally a portable dialysis machine?

I can agree with you on that however a dialysis machine only removes toxins from the blood like the blood filter. It in no way heals the damage done by those toxins like what your trying to do here.

> I think the biggest downside of this is the fact it requires a second person.

Cryo tube also requires a second person to close it but offers a ton of benefits for that small inconvenience, the same applies for the blood filter, the benefits of removing toxic reagents out ways the small inconvenience of requiring a 2nd person & so is a fair price to pay, i don't see how that is an argument here.

Overall just removing a constant -5 toxin damage regardless of the type of reagents present in the blood is just nonsensical. Amount & type of reagents present need to be considered in making such calculations & not a random constant of -5.

> I had the idea to make it do a small amount of O2 damage in trade

This doesn't make sense either. You need a reason on how & why the blood filter could cause this damage and not just for the sake of it. 

The cryo tube already does the job of healing any & all types of damage so even a dialysis machine wouldn't add much value to the game & same applies for your changes.

Reply:
The O2 damage comes from the concept of dialysis, since the blood isn't actually in your body while it's being cleased, it can lead to a hypoglycemia effect. The problem comes in during a blackout, if people are poisoned and there's no power or pre-prepared meds, there's very little you can do for toxins.